### PR TITLE
Implementating Scipy Wrapper and Pure Python for Bisection Algorithm

### PR DESCRIPTION
**Pull request by Tolani4**
## Linting
  - Hasn't been linted yet.

## Contents
This pull request introduces the implementation of the bisection method using a wrapper for Scipy's bisect function.
This pull request also implements the bisection algorithm using pure Python.
Files Changed:
calculus.py: Added the bisection_scipy and bisection_pure functions.
test_calculus.py: Added unit tests for bisection_scipy.

Assignment
Task Group 1 - Bisection Algorithm

Implementation Type: Using Scipy wrapper and only pythons cores features for implementation of the bisection algorithm.
Description: A Python function using Scipy's bisect to estimate roots of a continuous function over a specified interval. As well as an implementation of the bisection algorithm using only Python’s core features.

## Labels and Milestones
  - Label: Algorithm 
Milestone: Implementating Scipy Wrapper and Pure Python for Bisection Algorithm

## Reviewer
  - dhamalakamal

### CODE REVIEW

**Comment by zbpetersbuf** [comment]
@Tolani4 you still need a reviewer ill be your second, but you still need to fix up your pylon issues, such as your unused import pytest and trailing newlines you also need to add a milestone

File: test_calculus.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
 """
 test_calculus.py
 """
-import calculus as calc
```
Comment (laserlab) [inline test_calculus.py:4]: This will be break other implementations, just use this import statement

File: test_calculus.py [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+import pytest
+from calculus import bisection_scipy
```
Comment (laserlab) [inline test_calculus.py:5]: delete this one

**Review by laserlab — CHANGES_REQUESTED** [review]
- change import statement to not break existing code
- sync repo for linting

File: test_calculus.py (hunk 2 above)
Comment (Tolani4) [reply test_calculus.py:5]: I changed the import statements and tried to sync the repo for linting however the pylint review has been stuck on 
"Expected — Waiting for status to be reported" for a while, what course of action can I take.

**Comment by zbpetersbuf** [comment]
@Tolani4 have you synced your fork?

**Comment by Tolani4** [comment]
I did, all my branches are up to date however  pylint still isnt giving me a review.

**Comment by zbpetersbuf** [comment]
@Tolani4 see these two line in your test_calculus.py

from calculus import bisection_scipy
from calculus import bisection_pure
this is wrong it is incorrect, look back at previous comments on how to fix this

and since you deleted the dummy function and test_dummy this will cause conflicts that you must resolve

if you do not look back at the previous comments and change this im not sure how to help you

**Comment by zbpetersbuf** [comment]
maybe going to @laserlab s office may help save time and get this fixed

**Comment by laserlab** [comment]
> from calculus import bisection_scipy
> from calculus import bisection_pure

Yes please just import is as I commented. This will break other people code and is not necessary

**Comment by zbpetersbuf** [comment]
@Tolani4 the branch that it is on is 50 commits behind it is not synced

**Pull request closed without merging**